assets/component-linked-fields.js

```diff
@@ -0,0 +1,101 @@
+class LinkedFields extends HTMLElement {
+  static TYPES = ["country", "region", "city"];
+
+  constructor() {
+    super();
+    this.locale = document.documentElement.lang || "en";
+    this.countryCode = CUSTOMER_COUNTRY_CODE;
+    this.regionCode = null;
+
+    for (const name of LinkedFields.TYPES) {
+      this[name] = this.querySelector(`yc-combobox[name='${name}']`);
+    }
+  }
+
+  connectedCallback() {
+    this._render();
+  }
+
+  _render() {
+    this.fetchOptions();
+  }
+
+  async fetchOptions() {
+    for (const type of LinkedFields.TYPES) {
+      this[type] && (await this.onFetch(type));
+    }
+  }
+
+  setupOptions(type, options) {
+    const combobox = this[type];
+    const content = combobox.querySelector("yc-combobox-content");
+    const search = combobox.querySelector("input[type='search'");
+
+    content.innerHTML = "";
+    content.appendChild(search);
+
+    options.forEach((opt, index) => {
+      const isDefault = {
+        country: opt.code == this.countryCode,
+        region: index === 0,
+        city: index === 0,
+      };
+
+      content.insertAdjacentHTML(
+        "beforeend",
+        `<yc-combobox-item value="${typeof opt === "string" ? opt : opt.code}" ${isDefault[type] ? "checked" : ""}>
+            ${typeof opt === "string" ? opt : opt.name}
+        </yc-combobox-item>`,
+      );
+    });
+
+    combobox.setup(content.querySelectorAll("yc-combobox-item"), (e) => this.onChange(e, type));
+  }
+
+  async onChange(e, type) {
+    const value = e.target.value;
+
+    if (type === "country") this.countryCode = value;
+    if (type === "region") this.regionCode = value;
+    for (const next of { country: LinkedFields.TYPES.slice(1), region: LinkedFields.TYPES.slice(2) }[type] || []) {
```

Comment:
This is a bit complex, we might want to break it down, maybe a function that gets the "dependency fields" based on the type:

```js
getDependentFields(type) {
  if (type === "country") return LinkedFields.TYPES.slice(1);
  if (type === "region") return LinkedFields.TYPES.slice(2);
  return [];
}
```

```suggestion
    const dependentFields = this.getDependentFields(type);
    for (const fieldType of dependentFields) {
      await this.onFetch(fieldType);
    }
```

Reply:
😞

Reply:
https://youcan-shop.slack.com/archives/C04CYFXKBKN/p1746031159018939?thread_ts=1746012498.308409&cid=C04CYFXKBKN